### PR TITLE
chore: [App Sidebar] Data Pane

### LINKED ISSUE #27952: [Feature]: Add Settings pane
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

https://www.notion.so/appsmith/Unified-add-panel-18a94e51a498410aa5503379babb6ef0?p=257e76b5213944fe8dd144a8f1d3a9a5&pm=s

### Why should this be worked on?

part of unified add panel

### LINKED ISSUE #27956: [Feature]: Add Data pane with new url structure
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

https://www.notion.so/appsmith/Unified-add-panel-18a94e51a498410aa5503379babb6ef0?p=83b9fffaa7da47a2bd020aa06e5c95c3&pm=s

### Why should this be worked on?

part of unified add panel

### PR DESCRIPTION
**Pull request by hetunandu**
## Description
Works on the new data pane as part of the App Sidebar. This is currently behind a feature flag. 

- Shows workspace datasources on the left pane
- Updated create new flow by refactoring the existing IntegrationsHomeScreen
- Updates context switching for default value selectors
- Fixes some settings pane issues

#### PR fixes following issue(s)
Fixes #27956
Fixes #27952

#### Media

https://github.com/appsmithorg/appsmith/assets/12022471/29e36823-4469-418c-ad2a-5362caa93cf4

#### Type of change

- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
No new tests. Existing tests should pass

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/pages/Editor/DataSourceEditor/DatasourceBlankState.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import {
+  createMessage,
+  DATASOURCE_BLANK_STATE_MESSAGE,
+} from "@appsmith/constants/messages";
+import history from "utils/history";
+import { integrationEditorURL } from "@appsmith/RouteBuilder";
+import type { RouteComponentProps } from "react-router";
+import { INTEGRATION_TABS } from "constants/routes";
+
+const Container = styled.div`
```
Comment (albinAppsmith) [inline app/client/src/pages/Editor/DataSourceEditor/DatasourceBlankState.tsx:14]: Next time you can try using Flex component from ADS.
https://design-system.appsmith.com/?path=/story/design-system-flex--flex-story

File: app/client/src/pages/Editor/IDESidePane/DataSidePane.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+} from "@appsmith/constants/messages";
+
+const PaneContainer = styled.div`
+  width: 300px;
+`;
+
+const PaneHeader = styled.div`
+  height: 40px;
+  background: #f8fafc;
+  border-bottom: 1px solid #cdd5df;
```
Comment (albinAppsmith) [inline app/client/src/pages/Editor/IDESidePane/DataSidePane.tsx:31]: Please don't use custom colors... You can use tokens from ADS v2 eg: --ads-v2-color-border

**Review by albinAppsmith — APPROVED** [review]

File: app/client/src/pages/Editor/DataSourceEditor/DatasourceBlankState.tsx (hunk 1 above)
Comment (hetunandu) [reply app/client/src/pages/Editor/DataSourceEditor/DatasourceBlankState.tsx:14]: I tried this but the padding and gaps were not being applied for some reason

Comment (hetunandu) [reply app/client/src/pages/Editor/DataSourceEditor/DatasourceBlankState.tsx:14]: Yup thats how i did it as well.

Comment (albinAppsmith) [reply app/client/src/pages/Editor/DataSourceEditor/DatasourceBlankState.tsx:14]: Okay let me try it in main repo.

**Pull request merged**